sdk/program/src/entrypoint.rs

```diff
@@ -127,9 +127,17 @@ macro_rules! entrypoint {
         /// # Safety
         #[no_mangle]
         pub unsafe extern "C" fn entrypoint(input: *mut u8) -> u64 {
-            let (program_id, accounts, instruction_data) =
-                unsafe { $crate::entrypoint::deserialize(input) };
-            match $process_instruction(&program_id, &accounts, &instruction_data) {
+            use std::mem::MaybeUninit;
+            const UNINIT_ACCOUNT_INFO: MaybeUninit<AccountInfo> =
+                MaybeUninit::<AccountInfo>::uninit();
+            const MAX_ACCOUNT_INFOS: usize = 64;
```

Comment:
Just curious where the value 64 came from - is this just the stack size limit for `AccountInfo`?

Reply:
It's the current limit for accounts in an instruction: https://github.com/anza-xyz/agave/blob/2e6ca8c1f62db62c1db7f19c9962d4db43d0d550/runtime/src/bank.rs#L3219

Reply:
Insn't it 128?

https://github.com/anza-xyz/agave/blob/28cbac179ade62a69659067ebb13f71e54672d72/sdk/program/src/syscalls/mod.rs#L21-L25

Reply:
@joncinque I believe the value you've linked is for max number of write locks.

https://github.com/anza-xyz/agave/blob/28cbac179ade62a69659067ebb13f71e54672d72/accounts-db/src/accounts.rs#L512-L519

Reply:
@LucasSte the lock limit is configurable, but falls back to 128 when feature `increase_tx_account_lock_limit` is active. It's only active on devnet rn.

https://github.com/anza-xyz/agave/blob/2e6ca8c1f62db62c1db7f19c9962d4db43d0d550/runtime/src/bank.rs#L3213-L3218

Reply:
I'm seeing 255 as the limit for instructions.

https://github.com/anza-xyz/agave/blob/1871e2c70932d9fb8e31a6fcd4bdda2e05c25432/programs/bpf_loader/src/serialization.rs#L24-L26

https://github.com/anza-xyz/agave/blob/1871e2c70932d9fb8e31a6fcd4bdda2e05c25432/sdk/program/src/entrypoint.rs#L39-L40

Reply:
> I'm seeing 255 as the limit for instructions.

https://github.com/anza-xyz/agave/blob/28cbac179ade62a69659067ebb13f71e54672d72/sdk/program/src/syscalls/mod.rs#L16-L19

Reply:
It's 64, check out the test I added at [d56c0d6](https://github.com/anza-xyz/agave/pull/2618/commits/d56c0d6a6b92b68d5cb4708c641f77416bb908dd)

Reply:
Had to rebase, it's [3ad2cf9](https://github.com/anza-xyz/agave/pull/2618/commits/3ad2cf9e2c57279fae10fdac600355cfddf1b01a) now

Reply:
Nice, thanks for the tests! They make it pretty clear what the limit is now.

I'm curious how this doesn't break when the feature is enabled, if the allocation is set to 64?

Reply:
It's a pre-compiled program, so it's the entrypoint with allocations :sweat_smile: it's a good point though, if we replace that program with one compiled using the new `entrypoint`, it'll panic. 

I could also introduce `entrypoint_with_alloc` to use the old deserialization, but I don't want to encourage people to use it... what do you think?

Side note: I've added a better error message with 5abfb5f758

Reply:
> It's a pre-compiled program, so it's the entrypoint with allocations 😅 it's a good point though, if we replace that program with one compiled using the new `entrypoint`, it'll panic.

Ohh duh, lol. So activating that feature would effectively break programs using the no-alloc deserialization if they were to receive more than 64 accounts? Could you do 128 here instead - since you're trimming the slice anyway? Either way, this is kind of brittle because we can't feature-gate the value (no invoke context here).

> I could also introduce `entrypoint_with_alloc` to use the old deserialization, but I don't want to encourage people to use it... what do you think?

You don't want to encourage people to use which one - the new one or old one? I was thinking it might make more sense to do `entrypoint_no_alloc` for the new deserialization, since someone could blindly compile their program with `entrypoint` and not realize they're now using the new stuff, and could be restricted to 64 accounts post-feature activation.

Reply:
> Could you do 128 here instead - since you're trimming the slice anyway?

128 goes over the max per stack frame, since an array of 128 takes 6144 bytes

> You don't want to encourage people to use which one - the new one or old one?

I would discourage the old one, since it's less performant. My hope is that we can keep this until dynamic stack frames are enabled, at which point we should be able to bump it to 128 no problem. I don't know of the timeframe to increase the max account locks to 128.

The only potential problem would be a program that iterates through all remaining accounts, where it wouldn't be able to benefit from the increased account locks automatically. Seeing as that is a potential issue and could cause some breakage, I think you're right -- it would be safer to keep both implementations side by side. I'll do `entrypoint_no_alloc` and put the old one back in -- it's just annoying that programs won't be able to benefit from this unless they read the release notes.

Reply:
I've added back the old entrypoint and created a new one, let me know how it looks!